src/sparseml/optim/helpers.py

```diff
@@ -255,7 +255,9 @@ def _update_recipe_variable(var_name: str, var_value, container):
     key_found = False
 
     for key, value in container.items():
-        if not isinstance(value, list):
+        if not isinstance(value, list) or not all(
```

Comment:
Can we add some small unit tests for this just to be sure that this functionality isn't lost in the future?

Reply:
Added coverage to existing unit tests and confirmed it fails on main without this PR